Framework/Geometry/src/Objects/MeshObject.cpp

```diff
@@ -0,0 +1,725 @@
+#include "MantidGeometry/Objects/MeshObject.h"
+
+#include "MantidGeometry/Objects/Track.h"
+#include "MantidGeometry/Rendering/CacheGeometryHandler.h"
+#include "MantidGeometry/Rendering/GeometryHandler.h"
+#include "MantidGeometry/Rendering/vtkGeometryCacheReader.h"
+#include "MantidGeometry/Rendering/vtkGeometryCacheWriter.h"
+#include "MantidKernel/Exception.h"
+#include "MantidKernel/make_unique.h"
+#include "MantidKernel/Material.h"
+#include "MantidKernel/PseudoRandomNumberGenerator.h"
+
+#include <boost/make_shared.hpp>
+
+namespace Mantid {
+namespace Geometry {
+
+using Kernel::Material;
+using Kernel::V3D;
+using Kernel::Quat;
+
+/**
+*  Default constuctor
+*/
+MeshObject::MeshObject() : MeshObject("") {}
```

Comment:
Unless it is really necessary to have one, I would avoid defining a default constructor for this type.